### PR TITLE
Add StackBlitz example for test timeouts

### PR DESCRIPTION
**Pull request by SamVerschueren**
You can test the example on my fork by clicking the button

[Open in StackBlitz](https://stackblitz.com/github/SamVerschueren/ava/tree/stackblitz/examples/timeout/examples/timeouts?file=test.js&terminal=test&view=editor)

Example is pretty straight forward I think. Let me know if you want something differently.

### CODE REVIEW

File: examples/timeouts/test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+const test = require('ava');
+
+const {delay} = require('.');
```
Comment (novemberborn) [inline examples/timeouts/test.js:4]: What's the Node.js version used in Stackblitz? With 16 we can use `timers/promises`: https://nodejs.org/docs/latest-v16.x/api/timers.html#timers_timers_promises_api

File: examples/timeouts/test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	t.pass();
+});
+
+test('connect with database', async t => {
+	t.timeout(100, 'make sure database has started');
+
+	await delay(200);
+
+	t.pass();
+});
```
Comment (novemberborn) [inline examples/timeouts/test.js:20]: These kind of tasks you'd typically do in a `before` hook. Which should work with `t.timeout()` but that may be more confusing to people? So perhaps just make something up?

File: examples/timeouts/test.js (hunk 1 above)
Comment (SamVerschueren) [reply examples/timeouts/test.js:4]: We're currently on `14.x` unfortunately.

File: examples/timeouts/test.js (hunk 2 above)
Comment (SamVerschueren) [reply examples/timeouts/test.js:20]: Yes you're right. I'll change them.

Comment (SamVerschueren) [reply examples/timeouts/test.js]: Done. Hope this is a bit better :).

**Pull request merged**